awesome.c

```diff
@@ -543,6 +543,9 @@ main(int argc, char **argv)
     /* We have no clue where the input focus is right now */
     globalconf.focus.need_update = true;
 
+    /* set the default preferred icon size */
+    globalconf.preferred_icon_size = 0;
```

Comment:
The default is 0 which is pretty much what we are doing right now already (right now we pick the first one, now we will pick the smallest; however I only ever saw icons sorted by size and so this shouldn't make a difference).

The nice thing about 0 is that it's a DPI-independent value. :-)

I agree that "come up with something that makes us pick 'good' icons by default" is still left to do, but that is a hard problem and IMO shouldn't stop this PR.